browser/modules/SavantShieldStudy.jsm

```diff
@@ -176,4 +185,65 @@ class TelemetryEvents {
   }
 }
 
+class BookmarkObserver {
+  constructor(studyCategory) {
+    this.STUDY_TELEMETRY_CATEGORY = studyCategory;
+    // there are two probes: bookmark and follow_bookmark
+    this.METHOD_1 = "bookmark";
+    this.EXTRA_SUBCATEGORY_1 = "feature";
+    this.METHOD_2 = "follow_bookmark";
+    this.EXTRA_SUBCATEGORY_2 = "navigation";
+    // nsINavBookmarksService constant
+    this.TYPE_BOOKMARK = 1;
+  }
+
+  init() {
+    this.observer = {
```

Comment:
You may want to add skipTags: true, to the observer object, so you won't be notified of "fake" bookmarks we currently create for tags.

Reply:
Yes I do want that! Thank you; added.